WordPress/Classes/Utility/CoreDataQuery.swift

```diff
@@ -0,0 +1,146 @@
+import Foundation
+
+struct CoreDataQuery<Result: NSManagedObject> {
+    private var predicates = [NSPredicate]()
+    private var sortDescriptors = [NSSortDescriptor]()
+
+    private var includesSubentities: Bool = true
+
+    init() {
+    }
+
+    func ascending(by key: String) -> Self {
+        var query = self
+        query.sortDescriptors.append(NSSortDescriptor(key: key, ascending: true))
+        return query
+    }
+
+    func descending(by key: String) -> Self {
+        var query = self
+        query.sortDescriptors.append(NSSortDescriptor(key: key, ascending: false))
+        return query
+    }
+
+    func includesSubentities(_ value: Bool) -> Self {
+        var query = self
+        query.includesSubentities = value
+        return query
+    }
+
+    func count(in context: NSManagedObjectContext) -> Int {
+        (try? context.count(for: buildFetchRequest())) ?? 0
```

Comment:
It'll be a bit more work, but let's not eat the error here – we should pass it up to the caller to deal with (for now we can have the same implementation at a higher level, but there's a difference between "there are no cells" and "there was an error trying to fetch the data")